### ISSUE TITLE
[MSVC] constexpr functions that cannot be constexpr

### ISSUE THREAD

**Issue by shartte** [google/filament#517]
I am working my way through using Filament's public headers in a program compiled with MSVC, and run into various issues, mostly relating to constexpr:

One example:
```
math\tquathelpers.h(175): error C3615: constexpr function 'math::details::norm' cannot result in a constant expression
```

When removing constexpr, it compiles and works as expected.

I looked into the function, it calls `sqrt`, which is not constexpr in the standard. 
Is it sensible to declare this math::norm function constexpr, and why doesn't clang complain about it?

**Comment by shartte** [comment]
p.s.: in a more general sense. is it desired that Filament headers can be consumed by MSVC in general? This would make the prebuilt windows binaries alot more useful, since they could be used without having to use clang-cl.

**Comment by romainguy** [comment]
I think it would be great if the headers could be used with MSVC. I checked and clang's `sqrt` is not `constexpr` indeed. I have no idea why it works. @pixelflinger would probably know :)

**Comment by shartte** [comment]
I checked that it indeed works on Clang / GCC, and does not on MSVC.
Both Clang and GCC even emit assembly with the sqrt indeed precomputed, so simply removing constexpr here is not advisable.

Clang 6:
https://godbolt.org/z/xwCBtX

VS 2017.8:
https://godbolt.org/z/-_arDE

**Comment by shartte** [comment]
Fixed by #523
